src/Core/Checkout/Cart/CartRuleLoader.php

```diff
@@ -81,6 +82,15 @@ public function invalidate(): void
     private function load(SalesChannelContext $context, Cart $cart, CartBehavior $behaviorContext, bool $new): RuleLoaderResult
     {
         return Profiler::trace('cart-rule-loader', function () use ($context, $cart, $behaviorContext, $new) {
+            // If the processing starts with deferred errors already in the cart, the cart MUST be persisted
+            // to remove the errors from the stored cart
+            if (Feature::isActive('DEFERRED_CART_ERRORS')) {
+                $hasDeferredErrors = $cart->getErrors()->count() > 0;
+            } else {
+                $hasDeferredErrors = false;
+            }
```

Comment:
For the sake of Shopware codestyle and readability, try to avoid else branches.

```suggestion
            $hasDeferredErrors = false;

            // If the processing starts with deferred errors already in the cart, the cart MUST be persisted
            // to remove the errors from the stored cart
            if (Feature::isActive('DEFERRED_CART_ERRORS')) {
                $hasDeferredErrors = $cart->getErrors()->count() > 0;
            }
```

Reply:
Alternatively `$hasDefferedErrors = Feature::isActive('DEFERRED_CART_ERRORS') && $cart->getErrors()->count() > 0;`

Reply:
The real correct way would be:
```suggestion
            // If the processing starts with deferred errors already in the cart, the cart MUST be persisted
            // to remove the errors from the stored cart
            $hasDeferredErrors = $cart->getErrors()->count() > 0;
            
            if (!Feature::isActive('DEFERRED_CART_ERRORS')) {
                $hasDeferredErrors = false;
            }
```
Both of the other cases add noise to any deprecation removal PR. This is pretty much readable and will be easily removed.

Reply:
Fixed